flocker/node/_diagnostics.py

```diff
@@ -0,0 +1,320 @@
+        :returns: An absolute path string for a file inside the archive
+            directory.
+        """
+        return os.path.join(
+            self._archive_name,
+            name,
+        )
+
+    def _open_logfile(self, name):
+        """
+        :param str name: A unique label for the file.
+        :return: An open ``file`` object with a name generated by
+            `_logfile_path`.
+        """
+        return open(self._logfile_path(name), 'w')
+
+    def create(self):
+        """
+        Create the archive by first creating a uniquely named directory in the
+        current working directory, adding the log files and debug information,
+        creating a ``tar`` archive from the directory and finally removing the
+        directory.
+        """
+        os.makedirs(self._archive_path)
+        try:
+            # Flocker version
+            with self._open_logfile('flocker-version') as output:
+                output.write(__version__.encode('utf-8') + b'\n')
+
+            # Flocker logs.
+            services = self._service_manager.flocker_services()
+            for service_name, service_status in services:
+                self._log_exporter.export_flocker(
+                    service_name=service_name,
+                    target_path=self._logfile_path(service_name)
+                )
+
+            # Syslog.
+            self._log_exporter.export_all(self._logfile_path('syslog'))
+
+            # Status of all services.
+            with self._open_logfile('service-status') as output:
+                services = self._service_manager.all_services()
+                for service_name, service_status in services:
+                    output.write(service_name + " " + service_status + "\n")
+
+            # Docker version
+            check_call(
+                ['docker', 'version'],
+                stdout=self._open_logfile('docker-version')
+            )
+
+            # Docker configuration
+            check_call(
+                ['docker', 'info'],
+                stdout=self._open_logfile('docker-info')
+            )
+
+            # Kernel version
+            self._open_logfile('uname').write(' '.join(os.uname()))
+
+            # Distribution version
+            self._open_logfile('os-release').write(
+                open('/etc/os-release').read()
+            )
+
+            # Network configuration
+            check_call(
+                ['ip', 'addr'],
+                stdout=self._open_logfile('ip-addr')
+            )
+
+            # Hostname
+            self._open_logfile('hostname').write(gethostname() + '\n')
+
+            # Partition information
+            check_call(
+                ['fdisk', '--list'],
+                stdout=self._open_logfile('fdisk')
+            )
+
+            # Block Device and filesystem information
+            check_call(
+                ['lsblk', '--all'],
+                stdout=self._open_logfile('lsblk')
+            )
+
```

Comment:
Another thought for a later improvement: make each of these things data (e.g. a `Diagnostic` class with a `get` method). Really doesn't need to happen in this branch. :smile: